packages/network-controller/tests/helpers.ts

```diff
@@ -76,22 +83,56 @@ export const TESTNET = {
  * @returns The messenger.
  */
 export function buildRootMessenger(): RootMessenger {
-  return new Messenger();
+  return new Messenger({ namespace: 'Root' });
 }
 
 /**
- * Build a restricted messenger for the network controller.
+ * Build a messenger for the network controller.
  *
- * @param messenger - A messenger.
- * @returns The network controller restricted messenger.
+ * @param rootMessenger - The root messenger.
+ * @returns The network controller messenger.
  */
 export function buildNetworkControllerMessenger(
-  messenger = buildRootMessenger(),
+  rootMessenger = buildRootMessenger(),
 ): NetworkControllerMessenger {
-  return messenger.getRestricted({
-    name: 'NetworkController',
-    allowedActions: ['ErrorReportingService:captureException'],
-    allowedEvents: [],
+  const networkControllerMessenger = new Messenger<
+    'NetworkController',
+    AllNetworkControllerActions,
+    AllNetworkControllerEvents,
+    typeof rootMessenger
+  >({
+    namespace: 'NetworkController',
+    parent: rootMessenger,
+  });
+  rootMessenger.delegate({
+    messenger: networkControllerMessenger,
+    actions: ['ErrorReportingService:captureException'],
+  });
+  return networkControllerMessenger;
+}
+
+/**
+ * Build a messenger for the error reporting service.
+ *
+ * @param rootMessenger - The root messenger.
+ * @returns The error reporting service messenger.
+ */
+export function buildErrorReportingServiceMessenger(
+  rootMessenger = buildRootMessenger(),
+): Messenger<
+  'ErrorReportingService',
+  AllNetworkControllerActions,
+  AllNetworkControllerEvents,
+  typeof rootMessenger
+> {
+  return new Messenger<
+    'ErrorReportingService',
+    AllNetworkControllerActions,
+    AllNetworkControllerEvents,
```

Comment:
ErrorReportingService has its own actions and events, it doesn't use the ones from NetworkController. So it seems that this at least ought to be `AllErrorReportingServiceActions` and `AllErrorReportingServiceEvents`. Is this correct?